walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,123 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public static final int DAY_START = TimeRange.START_OF_DAY;
+    public static final int DAY_END = TimeRange.END_OF_DAY;
+    public static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+
+    /**
+      * Return a collection of TimeRanges that attendees can meet for a requested 
+      * meeting. If there are no TimeRanges that works for both required and optional 
+      * attendees, find TimeRanges that just work for required attendees. If there 
+      * are no required attendees, find the TimeRanges for just the optional attendees.
+      */
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        long meetingDuration = request.getDuration();
+        Collection<String> requiredAttendees = request.getAttendees(); 
+        Collection<String> allAttendees = new ArrayList<>(requiredAttendees);
+        allAttendees.addAll(request.getOptionalAttendees());
+
+        if (meetingDuration > WHOLE_DAY.duration() || meetingDuration <= 0) {
```

Comment:
I wonder if meetingDuration == 0 is also a valid request..

Reply:
Isn't `meetingDuration <= 0` covered `meetingDuration == 0 `?

Reply:
> A meeting request has:
a name
a duration in minutes
a collection of attendees

Since it's a duration, it shouldn't be 0 right?

Reply:
I think either way is fine. Just if it's a valid request, then the function should return all free slots, right? 
I agree with you and prefer to treat it as invalid request. 
Marking as resolved.